cumulusci/robotframework/tests/salesforce/ui.robot

```diff
@@ -49,6 +49,25 @@ Create test data
     ...                 ContactId=${CONTACT ID}
     Set suite variable  ${OPPORTUNITY ID}
 
+Object field should be
+    [Arguments]       ${obj name}  ${obj id}  ${field}  ${expected_value}
+    [Documentation]
+    ...  Fetches the object using the API, and verifies that the
+    ...  given field has the expected value
+
+    # it may take salesforce a second or two for the data
+    # to be saved and visible to the API, so we'll try this
+    # in a short loop
+    FOR  ${i}  IN RANGE  3
+        &{obj} =          Salesforce Get  ${obj name}  ${obj id}
+        ${actual_value}=  Set variable  ${obj}[${field}]
+
+        Return from keyword if  $expected_value == $actual_value
+        log  Retrying API call...  WARN
+        Sleep  1 second
+    END
+    Fail  Expected ${obj name} field ${field} to be '${expected_value}' but it was '${actual_value}'
+
```

Comment:
I'm surprised you didn't implement this one in Python.

Reply:
That might have been smarter. Though since at the moment the keyword is unique to this specific test case, it was simpler to do it here. "simpler" in that if it were in a library I'd have to write a test case for it. I think for now I'll leave it as is. 

It might be useful to make this (or something like it) an official keyword, but I'll save that for a separate work item. I'll ping the other robot users to see if they would find such a keyword useful.

Reply:
That makes sense that this is more of a one-off.